EventListener/TokenSubscriber.php

```diff
@@ -323,4 +333,123 @@ private function getCustomFieldValues(CustomObject $customObject, Token $token,
 
         return $fieldValues;
     }
+
+    public function onTokenReplacement(TokenReplacementEvent $event): void
+    {
+        $clickthrough = $event->getClickthrough();
+
+        if (!array_key_exists('dynamicContent', $clickthrough)) {
+            return;
+        }
+
+        $lead      = $event->getLead();
+        $tokenData = $clickthrough['dynamicContent'];
+
+        foreach ($tokenData as $data) {
+            $filterContent = $data['content'];
+
+            $isCustomObject = false;
+            foreach ($data['filters'] as $filter) {
+                $customFieldValues = $this->getCustomFieldDataForLead($filter['filters'], (int) $lead['id']);
+
+                if (!empty($customFieldValues)) {
+                    $isCustomObject = true;
+                    $lead           = array_merge($lead, $customFieldValues);
+                }
+
+                if ($isCustomObject && $this->matchFilterForLead($filter['filters'], $lead)) {
+                    $filterContent = $filter['content'];
+                    break;
+                }
+            }
+
+            if ($isCustomObject) {
+                $event->addToken('{dynamiccontent="'.$data['tokenName'].'"}', $filterContent);
+            }
+        }
+    }
+
+    /**
+     * @throws InvalidCustomObjectFormatListException
+     */
+    private function getCustomFieldValue(CustomObject $customObject, int $id, string $customFieldAlias): string
+    {
+        $orderBy  = CustomItem::TABLE_ALIAS.'.id';
+        $orderDir = 'DESC';
+
+        $tableConfig = new TableConfig(1, 1, $orderBy, $orderDir);
+        $tableConfig->addParameter('customObjectId', $customObject->getId());
+        $tableConfig->addParameter('filterEntityType', 'contact');
+        $tableConfig->addParameter('filterEntityId', $id);
+        $tableConfig->addParameter('token', $customFieldAlias);
```

Comment:
```suggestion
```
This line is unneeded. That means we are executing the same query:
```sql
SELECT CustomItem.*
FROM mautic_custom_item CustomItem
         LEFT JOIN mautic_custom_item_xref_contact CustomItemXrefContact
                   ON CustomItem.id = CustomItemXrefContact.custom_item_id
WHERE (CustomItem.custom_object_id = '1')
  AND (CustomItemXrefContact.contact_id = '2')
ORDER BY CustomItem.id DESC
LIMIT 1;
```
multiple times unnecessarily. For example if there are 3 filters, we execute that same query 3 times. Please avoid the unnecessary query executions.